Documentation/Yarn-Spec.md

```diff
@@ -0,0 +1,636 @@
+After encountering the header delimiter the body of the node is entered.
+
+#### Header Tags
+
+A *header tag* is a line broken up into three components, in order; the tag name, the separator, and the tag text.
+The *tag name* can be any combination of the characters `A` - `Z`, `a` - `z`, `0` - `9`.
+The *tag separator* is the character `:`.
+The *tag text* can be any combination of the characters `A` - `Z`, `a` - `z`, `0` - `9`.
+Header tags are commonly used as node specific metadata but using them in this manner is not required.
+
+The amount of allowed whitespace between the tag name, the seperator, and the tag text is unspecified.
+An example of a header tag is the title tag: `Title:start`.
+
+Every node must have a title tag.
+Required or banned header tags beyond title are unspecified.
+The order of header tags is unspecified.
+
+#### Title Tag
+
+The *title tag* is a specific header tag that uniquely identifies the node.
+The tag name for the title tag must be `Title`.
+The tag text for the title tag must be unique within the file.
+The tag text for the title tag should be unique within the project.
+
+The behaviour of the program when a title tag's text is not unique across the project is unspecified.
+The program could either flag this as an error and abort, choose which node will be the canonical one with that title, or some other form of name conflict resolution.
+
+### Body
+
+A body is comprised of multiple statements.
+A *statement* is a line that is one of the following:
+
+- dialogue
+- commands
+- options
+
+A statement may have optional hashtags.
+A body must have at least one statement.
+
+The body ends when encountering a line that contains the *body delimter* `===`.
+The body delimiter ends both the current node and the body of that node.
+The end of file must not be used in place of a body delimiter.
+
+#### Hashtags
+
+*Hashtags* are metadata associated with the statement they are a part of.
+Hashtags must go at the end of the statement.
+The other components of the statement must end at the hashtag, the hashtag operates effectively as the newline terminator for the statement.
+
+A hashtag starts with the `#` symbol and contain any text up to the newline or another hashtag.
+`#lineID:a10be2` is an example of a hashtag.
+Multiple hashtags must be separated by the `,` symbol.
+`#lineID:a10be2, #return` is an example of multiple hashtags on a line.
+
+### Dialogue Statement
+
+A dialogue statement is a statement that represents a single line of text in the yarn story.
+In most cases dialogue will be the bulk of the content of a node's body.
+Dialogue lines can be interpolated dialogue or raw dialogue.
+
+An *interpolated dialogue* is one where there are in-line expressions in the line.
+Expressions are encapsulated within the `{` and `}` symbols and it is the presence of these symbols that determine if a line is an interpolated one or not.
```

Comment:
This implies there's no way to insert `{` and `}` into dialogue. I would recommend that `\{` and `\}` are permitted as escaped characters